ferrocene/doc/user-manual/src/self-test/using.rst

```diff
@@ -0,0 +1,27 @@
+.. SPDX-License-Identifier: MIT OR Apache-2.0
+   SPDX-FileCopyrightText: The Ferrocene Developers
+
+Using
+=====
+
+``ferrocene-self-test`` ensures that the Ferrocene toolchain is installed properly.
+
```

Comment:
I think we should add CLI docs as well at `src/self-test/cli.rst`, to be consistent with `rustfmt` and `rustc`. Even if they currently are gonna be very short.

```suggestion

It accepts no command-line arguments, except `--help` and `--version`. See :doc:`the cli reference <cli>`.

```

Reply:
PS. Some automation to notify us when the cli reference goes out of sync with the output of `--help` would be cool. Or just notifying us when the `--help` output changes.

Reply:
the tool accepts no cli args, and we recently even made it [fail when any are provided](https://github.com/ferrocene/ferrocene/pull/907)

Reply:
Ah true, I forgot that input is provided through env variables. Then I think we should specify the environment variables the user can set on runtime: https://github.com/ferrocene/ferrocene/blob/6b20b454d0fe9377b9247d91b2680f47ccde6fef/ferrocene/tools/self-test/src/env.rs#L39-L41

Reply:
For this we probably have to add a `.. cli:env::` to the `cli:program` directive. Maybe just write it in text for this PR, and then we can adapt the directive in a follow-up PR.

Reply:
cbbe94bddb4